packages/react-composites/src/composites/ChatComposite/adapter/ChatAdapter.ts

```diff
@@ -15,6 +17,20 @@ export type ChatAdapterUiState = {
   // FIXME(Delete?)
   // Self-contained state for composite
   error?: Error;
+  /* @conditional-compile-remove-from(stable): FILE_SHARING */
+  /**
+   * Files being uploaded by a user in the current thread.
+   * Should be set to null once the upload is complete.
+   * @beta
+   */
+  uploadedFiles?: UploadedFile[];
+  /* @conditional-compile-remove-from(stable): FILE_SHARING */
+  /**
+   * Marks all the `uploadedFiles` as completely uploaded.
+   * Allows a message to be sent after the upload is complete.
+   * @beta
+   */
+  uploadedFilesCompleted?: boolean;
```

Comment:
instead of having an explicit boolean that must be maintained, can this be calculated from `uploadedFiles.<some accumulator>.isUploaded()`?

Reply:
e.g. could `getUploadedFilesCompleted` be something like 
```
getUploadedFilesCompleted => uploadedFiles.map(file => file.isUploaded()).reduce((a, b) => a && b)
```

Reply:
Been thinking the same thing @JamesBurnside  :D
Good suggestion.